src/headbar/headbar.router.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+angular.module('oscra-ui.headbar').config(function($stateProvider, $urlRouterProvider) {
+    $stateProvider
+        .state('root', {
+            url: '/root',
+            views:{
+                'headbar':{
+                    template: require('./headbarlayout.html'),
+                    controller:function($scope, $mdSidenav){
```

Comment:
make this an "require" and write your controller in another file. Don't use $scope